collect/central_collector.go

```diff
@@ -71,10 +84,31 @@ func (c *CentralCollector) Start() error {
 		return nil
 	})
 
+	c.eg.Go(func() (err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				err = fmt.Errorf("panic in central collector: %v\n%s", r, debug.Stack())
```

Comment:
I like the idea of catching the panic and dumping the stack -- but I think that what this means is that if one of these processes panics, it will die and leave the rest running. 

I think instead we want to think about restarting the dead task (so that we can recover from a error that's driven by bad data) and put a counter limit on it (so we won't keep trying forever). If the counter expires, we should shut down refinery completely so that our orchestrator can restart the whole process.

(Or alternatively, never retry and just rely on restarts.)